### ISSUE TITLE
[Bug]: Android High Contrast for Checkbox and Radio button

### ISSUE THREAD

**Issue by B3nz01d** [Orange-OpenSource/ouds-flutter#327]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

Samsung Galaxy A3 (And.8) - Sony Xperia XZ3 (And.10) - Huawei Mate 20 Pro (And.10) - Pixel 7 pro (Android 15)

### OS Version

see ref (Android version)

### App version

build 361

### Reference
this bug is linked to issue #155

### Describe the issue

None of the android OS allow to take into account the High contrast mode and change the behavior of the Checkbox and Radio button colors.

I've set up the following settings: 
#### Samsung Galaxy A3 (And.8)
* Settings > Accessibilité > Vue > Police contraste élevé -> on

#### Sony Xperia XZ3 (And.10)
* Settings > Accessibility > High Contrast text -> on

#### Huawei Mate 20 Pro (And.10)
* Settings > Accessibility features > Accessibility > High contrast text -> on

#### Pixel 7 pro (Android 15)
* Settings > Accessibility > Display size and text > High contrast text -> on

### Expected Behavior

The high contrast mode on android should be supported

**Comment by nouha06** [comment]
In **Flutter**, high contrast mode detection depends on the operating system:

- **iOS / macOS**: High contrast mode is properly detected when the user enables it.

- **Android**: The value always returns false. Therefore, an internal app parameter is required to handle contrast.

Flutter has not yet wired this functionality into MediaQuery.

To support high contrast text on Android, i propose two solutions :

1. Add a parameter to the Switch and Checkbox components, and use the demo application to manually handle contrast mode for Android.

1. Create a small Flutter plugin using MethodChannel to call AccessibilityManager.isHighTextContrastEnabled for Android.

cc @paulinea @B3nz01d @florentmaitre @AhmedAmineZr @Tayebsed93

**Comment by nouha06** [comment]
As discussed with the dev team, the plugin will be added to the core library.

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (434)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 382
cc @B3nz01d (product owner)
cc @Tayebsed93 @nouha06 @AhmedAmineZr (dev team)
cc @jerome-regnier (design team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (434)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 382
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by B3nz01d** [comment]
Closed during Sprint # 23

**Comment by B3nz01d** [comment]
@pya35 can you confirm this is ok?

**Comment by pya35** [comment]
> [@pya35](https://github.com/pya35) can you confirm this is ok?

Ok for radio, checkbox and switch, I'll raise a new issue for chip
